uniffi_bindgen/src/interface/function.rs

```diff
@@ -199,16 +199,16 @@ impl Argument {
         &self.name
     }
 
-    pub fn type_(&self) -> Type {
-        self.type_.clone()
+    pub fn type_(&self) -> &Type {
+        &self.type_
     }
 
     pub fn by_ref(&self) -> bool {
         self.by_ref
     }
 
-    pub fn default_value(&self) -> Option<Literal> {
-        self.default.clone()
+    pub fn default_value(&self) -> Option<&Literal> {
+        self.default.as_ref()
```

Comment:
Very nitty, but should this be `borrow()`?  I think of `as_ref()` as doing some sort of conversion between references.

Reply:
It can't be `borrow` because this is `Option::as_ref`, not `AsRef::as_ref`. `.borrow()` would result in `&Option<Literal>`, not `Option<&Literal>`.